core/contracts/Voting.sol

```diff
@@ -607,56 +534,41 @@ contract Voting is Testable, MultiRole, OracleInterface, VotingInterface, Encryp
-
-        for (uint i = 0; i < lastActiveVotingRound.priceRequestIds.length; i++) {
-            bytes32 priceRequestId = lastActiveVotingRound.priceRequestIds[i];
-            PriceRequest storage priceRequest = priceRequests[priceRequestId];
-
-            // TODO: we should probably take this assert out before we move to production to keep the voting
-            // contract from locking in the case of a bug. This would be an assert, but asserts don't allow
-            // messages.
-            require(priceRequest.lastVotingRound == lastActiveVotingRoundId,
-                "Found price request that was incorrectly placed in a round");
-
-            VoteInstance storage voteInstance = priceRequest.voteInstances[lastActiveVotingRoundId];
-
-            (bool isResolved, int price) = voteInstance.resultComputation.getResolvedPrice(
-                _computeGat(lastActiveVotingRoundId));
-
-            if (!isResolved) {
-                // If the vote cannot be resolved, push the request into the current round.
-                rounds[nextVotingRoundId].priceRequestIds.push(priceRequestId);
+        return round.snapshotId;
+    }
 
-                // Set the price resolution round number to the provided round.
-                priceRequest.lastVotingRound = nextVotingRoundId;
+    function _resolvePriceRequest(PriceRequest storage priceRequest, VoteInstance storage voteInstance) private {
+        if (priceRequest.isResolved) {
+            return;
+        }
+        (bool isResolved, int resolvedPrice) = voteInstance.resultComputation.getResolvedPrice(
+            _computeGat(priceRequest.lastVotingRound));
+        require(isResolved, "Can't claim reward for unresolved price request");
 
-                // Zero out the price request to reduce gas costs.
-                delete lastActiveVotingRound.priceRequestIds[i];
+        emit PriceResolved(priceRequest.lastVotingRound, priceRequest.identifier, priceRequest.time, resolvedPrice);
+        priceRequest.isResolved = isResolved;
 
-                // Delete the result computation since it's no longer needed.
-                delete voteInstance.resultComputation;
+        // Delete the resolved price request from pendingPriceRequests.
+        uint lastIndex = pendingPriceRequests.length - 1;
+        PriceRequest storage lastPriceRequest = priceRequests[pendingPriceRequests[lastIndex]];
+        lastPriceRequest.index = priceRequest.index;
+        pendingPriceRequests[priceRequest.index] = pendingPriceRequests[lastIndex];
+        delete pendingPriceRequests[lastIndex];
+        priceRequest.index = 0;
+    }
 
-                emit PriceRequestRolledOver(nextVotingRoundId, priceRequest.identifier, priceRequest.time);
-            } else {
-                emit PriceResolved(priceRequest.lastVotingRound, priceRequest.identifier, priceRequest.time, price);
-            }
+    function _updateRound(uint blockTime) private {
```

Comment:
You mentioned that the reason we're doing stateless time is because of the inflation rate. I'm assuming we *could* move to stateless time if we saved the inflation rate at the same time we saved the snapshot id? Because then we could use the 0 snapshot id to tell us whether to save it or not during reveal. The other thing we could do is put an init boolean in the Round data structure to denote whether the round has had at least one valid commit.  Am I missing something or would doing something like that allow us to move away from the stateful time system?

Note: I'm not suggesting we *should* do either of those, just trying to make sure I understand the reasoning/option space.

Reply:
Yeah that would work. It would change (IMO improve) the semantics of when inflation rate is captured and when changes take effect.